public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -879,6 +916,10 @@ public SurveyQuestionResult apply(AchillesResult o) {
         List<org.pmiops.workbench.model.QuestionConcept> mappedQuestions = mapAnalysesToQuestions(analyses, subIds, questions);
         org.pmiops.workbench.model.QuestionConcept questionConcept = mappedQuestions.get(0);
 
+        if (questionConcept == null) {
+            throw new DataNotFoundException("Unable to fetch results of this survey question");
+        }
```

Comment:
Can be
`Optional.ofNullable(questionConcept).orElseThrow(() -> new DataNotFoundException("Unable to fetch results of this survey question"));`

Reply:
Done thanks.